unit/test-grilrequest.c

```diff
@@ -1213,7 +1213,7 @@ static void test_request_dial(gconstpointer data)
 	enum ofono_clir_option clir = test_data->clir;
 	struct parcel rilp;
 
-	g_ril_request_dial(NULL, ph, clir, &rilp);
+	g_ril_request_dial(NULL, ph, FALSE, clir, &rilp);
```

Comment:
The problem is that the codepath where the emergency flag is being evaluated only activates when the plugin is detected. All current test cases there have `*gril` set to NULL, meaning it will never find a plugin type and also not switch to Samsung mode. To make better test cases this would involve mocking the result of the `g_ril_vendor(gril)` call which I am not able to do, honestly.